algorithms/java/BreadthFirstSearch.java

```diff
@@ -0,0 +1,58 @@
+// Illustration of Breadth First Search in java
+import java.io.*; 
+import java.util.*; 
+  
+class BSF 
+{ 
+    private int V;
+    private LinkedList<Integer> adj[];
+
+    Graph(int v) 
+    { 
+        V = v; 
+        adj = new LinkedList[v]; 
+        for (int i=0; i<v; ++i) 
+            adj[i] = new LinkedList(); 
+    } 
+ 
+    void addEdge(int v,int w) { 
+        adj[v].add(w); 
+    } 
+  
+    
+    void BFS(int s) { 
+        boolean visited[] = new boolean[V];        			 
+        LinkedList<Integer> queue = new LinkedList<Integer>(); 
+        visited[s]=true; 
+        queue.add(s); 
+        while (queue.size() != 0) { 
+            s = queue.poll(); 
+            System.out.print(s+" ");
+            Iterator<Integer> i = adj[s].listIterator(); 
+            while (i.hasNext()) { 
```

Comment:
If you are copying from another website then please provide the link to that website.

Reply:
This is mainly done by me, I just took an overview from gfg. Though I'll make the changes if you insist.

Reply:
Great Thanks for your contribution.